util/test/sub_test

```diff
@@ -182,6 +182,15 @@ def bytes_to_str(byte_string):
     else:
         return byte_string
 
+def concat_streams(stdout, stderr):
+    """ Concats stderr to stdout. Returns str is stdout is str, bytes otherwise """
+    if sys.version_info[0] >= 3:
+        if isinstance(bytes_to_str(stdout), str):
+            return bytes_to_str(stdout) + bytes_to_str(stderr)
```

Comment:
If stderr is binary data (bytes), will this fail because we're trying to do `str+bytes`?  (Trying to do the converse, `bytes+str`, was what caused the mandelbrot-fast failure that pointed out the flaw in my original "solution".)  While admittedly unlikely, binary `stderr` contents could result from error message corruption, which I have seen with mpirun-based launchers when programs use `fork(2)` via `Spawn`, so it's not impossible.  If this will indeed happen, maybe what we should do here is produce a `str` result if `str_to_bytes()` is a `str` for both `stdout` and `stderr`, and a `bytes` result otherwise.  (I'm assuming here that we can always produce `bytes` from a `str`, but sometimes can't produce a `str` from `bytes`.)  Otherwise, we'll get a python error here and that will obscure (lose altogether?) the `stderr` from the test itself.

Reply:
Yeah, my assumption was that stderr is always text. I'll update